domain/servicefactory/controller.go

```diff
@@ -191,5 +193,6 @@ func (s *ControllerFactory) SecretBackend() *secretbackendservice.WatchableServi
 func (s *ControllerFactory) Macaroon() *macaroonservice.Service {
 	return macaroonservice.NewService(
 		macaroonstate.NewState(changestream.NewTxnRunnerFactory(s.controllerDB)),
+		clock.WallClock,
```

Comment:
Maybe this should be injected via params even further?

Reply:
In a subsequent PR can we pass that through the service factory.